applications/luci-app-wireguard/luasrc/view/wireguard.htm

```diff
@@ -0,0 +1,281 @@
+      if not is_valid(transfer) then break end
+      local ln = string.split(strip(transfer), "\t")
+      peer_set_transfer(peers, ln[1], strip(ln[2]), strip(ln[3]))
+    end
+
+    table.insert(data, {
+      name = name,
+      public_key = strip(public_key),
+      listening_port = tonumber(strip(listening_port)),
+      peers = peers
+    })
+  end
+
+  if luci.http.formvalue("status") == "1" then
+    luci.http.prepare_content("application/json")
+    luci.http.write_json(data)
+    return
+  end
+-%>
+
+<%+header%>
+
+<script type="text/javascript" src="<%=resource%>/cbi.js"></script>
+<script type="text/javascript">//<![CDATA[
+
+  function bytes_to_str(bytes) {
+     var sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+     if (bytes == 0) return '0 Byte';
+     var i = parseInt(Math.floor(Math.log(bytes) / Math.log(1024)));
+     return Math.round(bytes / Math.pow(1024, i), 2) + ' ' + sizes[i];
+  };
+
+  function timestamp_to_str(timestamp) {
+    var now = new Date();
+    var seconds = (now.getTime() / 1000) - timestamp;
+    if (seconds < 60){
+      return parseInt(seconds) + '<%:s ago%>';
+    }
+    else if (seconds < 3600){
+      return parseInt(seconds / 60) + '<%:m ago%>';
+    }
+    else if (seconds < 86400){
+      return parseInt(seconds / 3600) + '<%:h ago%>';
+    } else {
+      return '<%:over a day ago%>';
+    }
+  }
+
+  XHR.poll(5, '<%=REQUEST_URI%>', { status: 1 },
+   function(x, data) {
+    for (var i = 0, ilen = data.length; i < ilen; i++) {
+      var iface = data[i];
+      var ifid = iface['public_key'] + "_";
+      var s = String.format(
+        '<strong><%:Public Key%>: </strong>%s' +
+        '<br /><strong><%:Listening Port%>: </strong>%<%:s%>',
+        iface['public_key'],
+        iface['listening_port']
+      );
+      document.getElementById(ifid + "info").innerHTML = s;
+      for (var j = 0, jlen = iface['peers'].length; j < jlen; j++) {
+        var peer = iface['peers'][j];
+        var pid = ifid + peer['public_key'] + "_";
+        s = String.format(
+          '<strong><%:Public Key%>: </strong>%s',
+          peer['public_key']
+        );
+        if (peer['endpoint'] != '(none)') {
+          s += String.format(
+            '<br /><strong><%:Endpoint%>: </strong>%s',
+            peer['endpoint']
+          );
+        }
+        if (peer['allowed_ips'].length > 0) {
+          s += '<br /><strong><%:Allowed IPs%>:</strong>';
+          for (var k = 0, klen = peer['allowed_ips'].length; k < klen; k++) {
+            s += '<br />&nbsp;&nbsp;&bull;&nbsp;' + peer['allowed_ips'][k];
+          }
+        }
+        if (peer['persistent_keepalive'] != 'off') {
+          s += String.format(
+            '<br /><strong><%:Persistent Keepalive%>: </strong>%ss',
+            peer['persistent_keepalive']
+          );
+        }
+        var icon = '<img src="<%=resource%>/icons/tunnel_disabled.png" />';
+        if (peer['latest_handshake'] > 0) {
```

Comment:
Ok, got it, so the 20 second difference between 2 minutes and 140 seconds is to be very sure the peer if offline. Thanks for the explanation!